sqlx-postgres/src/types/bigdecimal.rs

```diff
@@ -138,12 +138,18 @@ impl TryFrom<&'_ BigDecimal> for PgNumeric {
     }
 }
 
-/// ### Panics
-/// If this `BigDecimal` cannot be represented by `PgNumeric`.
 impl Encode<'_, Postgres> for BigDecimal {
     fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> IsNull {
+        use std::str::FromStr;
+        // If the argument is too big, then we replace it with a less big argument.
+        // This less big argument is already outside the range of allowed PostgreSQL DECIMAL, which
+        // means that PostgreSQL will return the 22P03 error kind upon receiving it. This is the
+        // expected error, and the user should be ready to handle it anyway.
         PgNumeric::try_from(self)
-            .expect("BigDecimal magnitude too great for Postgres NUMERIC type")
+            .unwrap_or_else(|_| {
+                PgNumeric::try_from(&BigDecimal::from_str(&format!("{:030000}", 0)).unwrap())
```

Comment:
This is a really clever way to trigger an error but if the only point is to generate a `scale` outside `0x3FFF` you can do it like so:

```rust
PgNumeric {
    digits: vec![0],
    scale: 0x400,
    weight: 0
}
```

Reply:
I took the liberty of making the change myself, as well as adding documentation covering this behavior and for `rust_decimal` as well.